### PR TITLE
[RFC] Refactor the Omise Swift architecture

### PR DESCRIPTION
**Pull request by pitiphong-p**
This PR refactors the overall Omise Swift architecture by adopting value type instead of the class based implementation.

I suggest the reviewer to pull this PR to your machine and review on the Xcode

### CODE REVIEW

File: OmiseSwift/APIRequest.swift [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import Foundation
+
+
+
+public class Request<Result: OmiseObject> {
```
Comment (chakrit) [inline OmiseSwift/APIRequest.swift]: Should be named `APIRequest` pa? (same as filename)

File: OmiseSwift/APIRequest.swift [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        let result: Failable<Result>
+        do {
+            switch httpResponse.statusCode {
+            case 400..<600:
+                let err: APIError = try deserializeData(data)
+                result = .fail(.api(err))
+                
+            case 200..<300:
+                result = .success(try endpoint.deserialize(data))
+//                if let resource = result as? ResourceObject {
```
Comment (igroomgrim) [inline OmiseSwift/APIRequest.swift]: Unused comment ?

**Review by igroomgrim — APPROVED** [review]

File: OmiseSwift/APIRequest.swift (hunk 1 above)
Comment (pitiphong-p) [reply OmiseSwift/APIRequest.swift]: Yeah you're right.
Renamed

File: OmiseSwift/APIRequest.swift (hunk 2 above)
Comment (pitiphong-p) [reply OmiseSwift/APIRequest.swift]: Oh thank you for pointing this for me
Removed

**Review by chakrit — APPROVED** [review]

**Pull request merged**